src/main/java/com/ministore/api/controllers/GetUserHandler.java

```diff
@@ -23,7 +26,13 @@ public GetUserHandler(UserRepository repository) {
         produces = "application/json"
     )
     public Optional<User> getUser(@PathVariable String id) {
-        return this.repository.findById(id);
+        Optional<User> user = this.repository.findById(id);
+
+        if (user.isEmpty()) { 
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "entity not found");
+        }
+
+        return user;
```

Comment:
There is an interesting case here.
`this.repository.findById()` returns an `Option<User>`, but if you are always checking if the `user.isEmpty` before returning, then `getUser()` doesn't really return an `Option<User>` anymore. It either returns an `User` or throws the exception.

It is important to change the return type of `getUser()` to `User`, because this signals to callers of this function that they don't need to deal with the case where this funcition returns an empty `Option`.

Reply:
Awesome! I applied your suggestion from the previous comment about using `.orElseThrow` and implemented the same approach here.